src/codecache/CodeCache.cpp

```diff
@@ -25,205 +25,230 @@
 #include "codecache/CodeCacheReaderWriter.h"
 #include "parser/Script.h"
 #include "parser/CodeBlock.h"
+#include <dirent.h>
+#include <sys/stat.h>
+#include <unistd.h>
+
+#define CODE_CACHE_FILE_DIR "/Escargot-cache/"
+#define CODE_CACHE_LIST_FILE_NAME "cache_list"
 
 namespace Escargot {
 
-#if defined(ESCARGOT_ENABLE_TEST)
-const char* CodeCache::CodeCacheContext::filePathPrefix = "";
-#else
-const char* CodeCache::CodeCacheContext::filePathPrefix = "/tmp/";
-#endif
+void CodeCacheContext::reset()
+{
+    m_cacheFilePath.clear();
+    m_cacheEntry.reset();
+
+    delete m_cacheStringTable;
+    m_cacheStringTable = nullptr;
+    m_cacheDataOffset = 0;
+}
 
-void CodeCache::CodeCacheContext::initStringTable()
+CodeCache::CodeCache()
+    : m_cacheWriter(nullptr)
+    , m_cacheReader(nullptr)
+    , m_enabled(false)
+    , m_modified(false)
 {
-    ASSERT(!this->stringTable);
+    initialize();
+}
 
-    this->stringTable = new CacheStringTable();
+CodeCache::~CodeCache()
+{
+    clear();
 }
 
-void CodeCache::CodeCacheContext::initStringTable(CacheStringTable* table)
+void CodeCache::initialize()
 {
-    ASSERT(!!table);
-    ASSERT(!this->stringTable);
+    // set cache file path
+    const char* homeDir = getenv("HOME");
```

Comment:
How about get cache dir when init VMIntance?
there are a few Escargot instances can exist(tizen.. android)

Reply:
Thank you for the review.
I've changed it to get the cache directory during VMInstance initialization and then pass it to CodeCache constructor.